### PR TITLE
fix(lambda): return layer permission from addPermission (#15524)

### PR DESCRIPTION
**Pull request by craigandrews**
When adding a Lambda layer permission, return the created `CfnLayerVersionPermission` as an escape hatch so the user can perform additional work on it. Discussed in #15524 

The use case in mind is to allow addition of retention policies without having to change the default behavior or forcing the user to replicate the call to the `CfnLayerVersionPermission` constructor each time.

----

### All Submissions:

* [X] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by craigandrews** [comment]
Closed pull request while further issues fixed. Primary issue is that `build.sh` doesn't work on MacOS if anything is inside a directory with a space (e.g. `~/Library/Application Support`)

**Comment by craigandrews** [comment]
Re-opened pull request after fixing problems with:

- shell escaping of commands with spaces
- no-op implementation of addPermission in KubectlLayer

**Comment by mrgrain** [comment]
Thanks for your contribution @craigandrews 

Would you mind splitting the build tools fix into a separate PR. That makes the handling much easier for us, e.g. if we need to rollback.

**Comment by craigandrews** [comment]
> Thanks for your contribution @craigandrews
> 
> Would you mind splitting the build tools fix into a separate PR. That makes the handling much easier for us, e.g. if we need to rollback.

Not at all! I've removed that commit from this PR, and created a new one for the shell escape here: https://github.com/aws/aws-cdk/pull/21267

File: packages/@aws-cdk/aws-eks-legacy/lib/kubectl-layer.ts [hunk 1]
```diff
@@ -68,8 +68,4 @@ export class KubectlLayer extends Resource implements lambda.ILayerVersion {
 
     this.layerVersionArn = Token.asString(resource.getAtt('Outputs.LayerVersionArn'));
   }
-
-  public addPermission(_id: string, _permission: lambda.LayerVersionPermission): void {
-    return;
-  }
```
Comment (mrgrain) [inline packages/@aws-cdk/aws-eks-legacy/lib/kubectl-layer.ts:74]: This removal seems like a rather large change. Why did `KubectlLayer` not support adding permissions and how come that it does support that now?

**Review by mrgrain — CHANGES_REQUESTED** [review]

**Comment by mrgrain** [comment]
Hey @craigandrews I've discussed this with the team now, and we generally avoid exposing L1 Constructs (like  `CfnLayerVersionPermission`) from L2.  I known it's slightly more code to write, but as you pointed out we can already achieve the same result by having the user create a `CfnLayerVersionPermission` themselves. Or they could locate the created node via `layerVersion.node.findChild()`.

File: packages/@aws-cdk/aws-eks-legacy/lib/kubectl-layer.ts (hunk 1 above)
Comment (craigandrews) [reply packages/@aws-cdk/aws-eks-legacy/lib/kubectl-layer.ts:74]: FWIW the only issue here is that changing the return type would not work for this scenario. I also considered having this function throw an error indicating that permissions were not supported.

**Pull request closed without merging**